File: apps/backend/src/account/Account.ts
import {
  Column,
  CreateDateColumn,
  Entity,
  ManyToOne,
  PrimaryGeneratedColumn,
} from 'typeorm';
import { AccountType } from './AccountType';
import { UUID } from '@mid/shared';
import { User } from '../user/User';

@Entity()
export class Account {
  @PrimaryGeneratedColumn('uuid')
  id: UUID;

  @Column({ type: 'varchar', length: 200 })
  name: string;

  @Column({ type: 'varchar', length: 200, nullable: true })
  description?: string;

  @Column({ type: 'varchar', length: 200 })
  issuer: string;

  @Column({ type: 'varchar', length: 200 })
  number: string;

  @Column({ type: 'numeric', precision: 5, scale: 2 })
  interestRate: number;

  @Column({ type: 'numeric', precision: 15, scale: 2 })
  withdrawalLimit: number;

  @CreateDateColumn({ type: 'timestamp', default: () => 'CURRENT_TIMESTAMP' })
  createdAt: Date;

  @ManyToOne(() => AccountType, {
    eager: true,
    nullable: false,
  })
  accountType: AccountType;

  @ManyToOne(() => User, {
    eager: true,
    nullable: false,
  })
  user: User;
}

File: backend/src/main/kotlin/com/kyoulho/mid/account/entity/Account.kt
package com.kyoulho.mid.account.entity

import com.kyoulho.mid.user.entity.MidUser
import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import java.time.LocalDateTime

@Entity
data class Account(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: String? = null,

    @Column(length = 200, nullable = false)
    var name: String,

    @Column(length = 200, nullable = false)
    var description: String?,

    @Column(length = 200, nullable = false)
    var issuer: String,

    @Column(length = 200, nullable = false)
    var number: String,

    @Column(nullable = false)
    var interestRate: String,

    @Column(nullable = false)
    var withdrawalLimit: String,

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    val createdAt: LocalDateTime = LocalDateTime.now(),

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "account_type_id")
    var accountType: AccountType,

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "user_id")
    val user: MidUser
)
